rules/alerting.go

```diff
@@ -412,6 +395,9 @@ func (r *AlertingRule) Eval(ctx context.Context, ts time.Time, query QueryFunc,
 		}
 	}
 
+	r.mtx.Lock()
```

Comment:
Moving it here LGTM. I was just wondering why we were locking above even if unnecessary 🤔